Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java

```diff
@@ -15,78 +15,60 @@
 
 import java.util.Objects;
 
 public class AddTagActivity extends AppCompatActivity implements TextWatcher {
-    private Bucket<Tag> mBucketTag;
-    private Button mButtonNegative;
-    private Button mButtonPositive;
-    private TextInputEditText mTagInput;
-    private TextInputLayout mTagLayout;
+    private AddTagViewModel viewModel;
+    private ActivityTagAddBinding binding;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         ThemeUtils.setTheme(this);
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_tag_add);
+        binding = ActivityTagAddBinding.inflate(getLayoutInflater());
 
-        int widthScreen = getResources().getDisplayMetrics().widthPixels;
-        int widthMargin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 56, getResources().getDisplayMetrics());
-        int widthMaximum = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 312, getResources().getDisplayMetrics());
-        LinearLayout layout = findViewById(R.id.layout);
-        layout.getLayoutParams().width = Math.min(widthMaximum, widthScreen - widthMargin);
-        layout.requestLayout();
-
-        ((View) layout.getParent()).setOnClickListener(
-            new View.OnClickListener() {
-                @Override
-                public void onClick(View view) {
-                    finishAfterTransition();
-                }
-            }
-        );
-        layout.setOnClickListener(null);
-
-        MorphSetup.setSharedElementTransitions(this, layout, getResources().getDimensionPixelSize(R.dimen.corner_radius_dialog));
+        Bucket<Tag> mBucketTag = ((Simplenote) getApplication()).getTagsBucket();
+        ViewModelFactory viewModelFactory = new ViewModelFactory(mBucketTag, this, null);
+        ViewModelProvider viewModelProvider = new ViewModelProvider(this, viewModelFactory);
+        viewModel = viewModelProvider.get(AddTagViewModel.class);
 
-        mBucketTag = ((Simplenote) getApplication()).getTagsBucket();
+        setObservers();
+        setupLayout();
+        setupViews();
 
-        TextView title = findViewById(R.id.title);
-        title.setText(getString(R.string.add_tag));
-
-        mTagLayout = findViewById(R.id.tag_layout);
-        mTagInput = findViewById(R.id.tag_input);
-
-        if (mTagInput != null) {
-            mTagInput.addTextChangedListener(this);
+        setContentView(binding.getRoot());
+    }
 
-            new Handler().postDelayed(
+    private void setupViews() {
+        binding.title.setText(getString(R.string.add_tag));
+        binding.tagInput.addTextChangedListener(this);
+        new Handler().postDelayed(
                 new Runnable() {
                     @Override
                     public void run() {
-                        mTagInput.requestFocus();
-                        DisplayUtils.showKeyboard(mTagInput);
+                        viewModel.showKeyboard();
```

Comment:
maybe use a different method name? the activity doesn't need to know that this triggers `showKeyboard`. Maybe something like `onAnimationFinished` or `onViewLoaded` or anything that sounds right

Reply:
This was changed to `viewModel.start()` to represent a loaded or started state in the UI. Fixed in ff0380dcc28559b1f4652e58177aa23988a8e8e0.